src/server/src/classes/Action.ts

```diff
@@ -63,55 +50,72 @@ export class Action {
     return true;
   }
 
-  call(): boolean {
+  bet(): boolean {
+    const betAmount = this.action.betAmount as number;
+    const callAmount = this.round.highestBet - this.player.currentBet;
+
+    const mainCases = (): boolean => {
+      if (betAmount == callAmount) {
+        return this.call();
+      } else {
+        if (callAmount <= 0) {
+          console.log(`player ${this.player.id} betting amount: ${betAmount}`);
+          this.placeBet(betAmount);
+          return true;
+        } else {
+          return this.raise();
+        }
+      }
+    };
+
+    // A bit hackey but actually works really well for our needs
+    switch (true) {
+      case betAmount > this.player.chipCount:
+        return false;
+      case betAmount == this.player.chipCount:
+        return this.allIn();
+      case betAmount < callAmount:
+        return false;
+      case betAmount < 0:
+        return false;
+      case betAmount == 0:
+        return this.check();
```

Comment:
technically, wouldn’t need to return check as an available action to the client if this is the case. but i think the best behavior here is to return bet(0) as false as an invalid action.